--- gst_wrapper.py
+++ gst_wrapper_pylon.py
@@ -247,6 +247,9 @@
             source = 'image'
     elif (input.source == 'videotestsrc'):
         source = 'videotestsrc'
+    # gst-plugin-pylon
+    elif (input.source.startswith('pylonsrc')):
+      source = 'pylonsrc'
     else:
         status = 'no file'
 
@@ -308,6 +311,33 @@
                                                      (input.width, input.height)
             source_cmd += ' tiovxcolorconvert ! video/x-raw, format=NV12 ! '
 
+    # gst-plugin-pylon
+    elif (source == 'pylonsrc'):
+        #Example: configuring to 640x480 @ 10fps in Mono8 format:
+            #gst-launch-1.0 pylonsrc ! "video/x-raw,width=640,height=480,framerate=10/1,format=GRAY8" ! videoconvert ! queue ! kmssink
+        
+        #Example: configuring to 640x480 @ 10fps in BayerRG8 format
+            #gst-launch-1.0 pylonsrc ! "video/x-bayer,width=640,height=480,framerate=10/1,format=rggb" ! bayer2rgb ! videoconvert ! queue ! kmssink
+
+        if (input.sen_id == 0):
+            source_cmd = 'pylonsrc ! '
+            
+        else:
+            source_cmd = 'pylonsrc device-serial-number=%d ! ' % input.sen_id
+        
+        if (input.format == 'bggr' or   # BayerBG8
+            input.format == 'grbg' or   # BayerGR8
+            input.format == 'rggb' or   # BayerRG8
+            input.format == 'gbrg'):    # BayerGB8
+                
+            source_cmd += 'video/x-bayer,width=%d,height=%d,framerate=%d/1,format=%s ! bayer2rgb ! ' % \
+                                                     (input.width, input.height, input.fps, input.format)
+        else:
+            source_cmd += 'video/x-raw,width=%d,height=%d,framerate=%d/1,format=%s ! ' % \
+                                                     (input.width, input.height, input.fps, input.format)
+
+        source_cmd += 'videoconvert ! queue ! '
+
     elif (source == 'http'):
         if not(input.format in video_dec):
             input.format = "auto"
@@ -345,6 +375,10 @@
                                        (input.width, input.height, input.format)
         if input.format != 'NV12':
             source_cmd += ' tiovxdlcolorconvert ! video/x-raw, format=NV12 ! '
+
+    # gst-plugin-pylon
+    elif (input.source.startswith('pylonsrc')):
+      source = 'pylonsrc'
 
     return source_cmd
 
